app/controllers/api/news_controller.rb

```diff
@@ -1,60 +1,31 @@
 # This is the new version of the API endpoint to retrieve NewsItems
 # based on a more sane query parameter setup.
 class Api::NewsController < ApiController
-  before_action :detect_news_feature
+  before_action :detect_news_feature, except: :index
 
   def index
     # Ensure fresh response on every request
     headers['Last-Modified'] = Time.now.httpdate
 
     distribute_reads(max_lag: MAX_ACCEPTABLE_REPLICATION_LAG, lag_failover: true) do
-      if params[:coinSlugs]
-        coins = Coin.where(slug: params[:coinSlugs])
+      news_items = nil
+
+      if params[:frontPage].present? # For Front page (public; no feature flag)
+        news_items = NewsItem.published.front_page
+          .includes(:coins, :news_categories)
+          .order_by_published
+          .limit(5)
+      else # For newsfeed (keep feature flag)
+        has_news = detect_news_feature
```

Comment:
I think `has_news` sounds like it would be a `boolean` but in this case is actually a json_response. Maybe we can change the name of this variable to be a little clearer